lib/board.rb

```diff
@@ -39,33 +39,40 @@ def consecutive_letters(letters, numbers)
   end
 
   def placement_fails(ship, coordinates)
-    coordinates.each do |coordinate|
-      if !valid_coordinate?(coordinate) || !@cells[coordinate].empty? || !length_check(ship, coordinates)
-        return true
+    if length_check(ship, coordinates) == false
```

Comment:
Nice! I think this makes more sense, to have the length check outside of the array iteration.